joinservice/cmd/main.go

```diff
@@ -47,6 +49,9 @@ func main() {
 	verbosity := flag.Int("v", 0, logger.CmdLineVerbosityDescription)
 	flag.Parse()
 
+	ctx, cancel := context.WithTimeout(context.Background(), vpcIPTimeout)
+	defer cancel()
```

Comment:
Moving this ctx up and using it for another function call goes aginst the original intention of this ctx. Either we should simply create a new `constext.Background` from which we later create a context with timeout OR we change the name of the timeout const.